File: sorting-algorithms/insertion-sort/insertionSort.py
import sys


def insertionSort(arr):
    """
    Sorts the given array using the insertion sort algorithm.

    Parameters:
    arr (list): The array to be sorted.

    Returns:
    list: The sorted array.
    """
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > temp:
            arr[j+1] = arr[j]
            j = j-1
        arr[j+1] = temp
    return arr


if __name__ == "__main__":
    a = list(map(int, sys.argv[1].strip("[]").split(",") ))
    print(insertionSort(a))

File: sorting-algorithms/insertion-sort/insertionSort.js
/**
 * Sorts an array using the insertion sort algorithm.
 * 
 * @param {Array} arr - The array to be sorted.
 * @returns {Array} - The sorted array.
 */
function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let temp = arr[i]
        let j = i - 1;

        while (j >= 0 && arr[j] > temp) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = temp;
    }
    return arr;
}

// Command line testing of insertionSort() using Node.js

// Parsing the command line argument as an array
let a = process.argv[2].slice(1, -1).split(',').map(Number);

// Sorting the array using the insertionSort() function
console.log(insertionSort(a));